src/main/java/mate/academy/spring/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,30 @@
+package mate.academy.spring.security;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userService.findByEmail(username)
+                .orElseThrow(() -> new UsernameNotFoundException("User not found"));
+        org.springframework.security.core.userdetails.User.UserBuilder builder =
+                org.springframework.security.core.userdetails.User.withUsername(username);
+        builder.password(user.getPassword());
+        builder.authorities(user.getRoles().stream()
```

Comment:
```suggestion
        builder.roles(user.getRoles().stream()
```
we talked about the difference